scripts/lua/policies/backendRouting.lua

```diff
@@ -53,7 +53,9 @@ function _M.setDynamicRoute(obj)
       u = url.parse(utils.concatStrings({'http://', dynamicBackend}))
     end
     if utils.tableContains(whitelist, u.host) then
-      ngx.req.set_uri(getUriPath(u.path))
+      if header:lower() ~= 'x-cf-forwarded-url' then
```

Comment:
This should be generic / configurable as part of the policy (i.e. it's not a `cf` policy)